### PR TITLE
remove uses of deprecated SL Combatant functions

### PR DESCRIPTION
**Pull request by ToppleTheNun**
convert uses of has2Piece to has2PieceByTier

convert uses of has4Piece to has4PieceByTier

replace ConduitLink with SpellLink

replace ConduitSpellText with BoringSpellValueText

replace all usages of Combatant#hasCovenant with false, then simplify

replace all usages of Combatant#hasSoulbind with false, then simplify

replace all usages of Combatant#hasConduit with false, then simplify

replace all usages of Combatant#hasCovenant with false, then simplify

replace all usages of Combatant#conduitRankBySpellID with 0, then simplify

### CODE REVIEW

**Comment by Sharrq** [comment]
Im ok with the changes to the mage specs. I have been working through the specs and removing those references, but have not quite made it through all of them yet.

**Review by Sharrq — APPROVED** [review]
I approve the mage spec changes

**Comment by oisinod** [comment]
@ToppleTheNun Priest stuff LGTM 👍

File: src/analysis/retail/monk/mistweaver/modules/features/Buffs.tsx [hunk 1]
```diff
@@ -44,17 +43,17 @@ class Buffs extends CoreAuras {
...
       {
         spellId: SPELLS.WEAPONS_OF_ORDER_BUFF_AND_HEAL.id,
-        enabled: combatant.hasCovenant(COVENANTS.VENTHYR.id),
+        enabled: false,
         timelineHighlight: true,
       },
       {
         spellId: SPELLS.BONEDUST_BREW_CAST.id,
-        enabled: combatant.hasCovenant(COVENANTS.VENTHYR.id),
+        enabled: false,
```
Comment (vohrr) [inline src/analysis/retail/monk/mistweaver/modules/features/Buffs.tsx]: `enabled: combatant.hasTalent(TALENTS_MONK.BONEDUST_BREW_TALENT),`

File: src/analysis/retail/monk/shared/TouchOfDeath.js [hunk 2]
```diff
@@ -27,7 +27,7 @@ class TouchOfDeath extends ExecuteHelper {
     options.abilities.add({
       spell: SPELLS.TOUCH_OF_DEATH.id,
       category: SPELL_CATEGORY.COOLDOWNS,
-      cooldown: this.selectedCombatant.hasLegendary(SPELLS.FATAL_TOUCH) ? 60 : 180,
+      cooldown: 180,
```
Comment (vohrr) [inline src/analysis/retail/monk/shared/TouchOfDeath.js]: `cooldown: 180 - 45 * this.selectedCombatant.getTalentRank(TALENTS_MONK.FATAL_TOUCH_TALENT.id),`

**Review by vohrr — CHANGES_REQUESTED** [review]
Couple of Mistweaver Suggestions

**Comment by trevorm4** [comment]
Monk looks good after the changes

**Review by emallson — APPROVED** [review]
I could've sworn i had updated those brew talents, but apparently i moved them and forgot to actually complete the updates 🤔

**Comment by tyndivelspaz** [comment]
Fury warrior and rogue looking good to me

**Review by Khazakdk — APPROVED** [review]
Changes to DK modules look good

**Pull request merged**